tslib.es6.js

```diff
@@ -348,6 +348,23 @@ export function __disposeResources(env) {
     return next();
 }
 
+export function __rewriteRelativeImportExtension(path, preserveJsx) {
+    if (typeof path === "string" && path[0] === "." && (path[1] === "/" || path[1] === "." && path[2] === "/")) {
+        if (path.substring(path.length - 4) === ".tsx") {
+            return path.substring(0, path.length - 4) + (preserveJsx ? ".jsx" : ".js");
+        }
+        if (path.substring(path.length - 3) === ".ts") {
+            var dot = path.lastIndexOf(".", path.length - 4);
+            if (dot >= 0 && (path.substring(dot - 2, dot) === ".d" || path.substring(dot, dot + 2) === ".d")) {
+                return path;
+            }
+            return path.substring(0, path.length - 3) + ".js";
+        }
+        return path.replace(/(?<!\.d)\.[cm]ts$/, function (ext) { return ext === ".mts" ? ".mjs" : ".cjs"; });

```

Comment:
Lookbehind assertions weren't added until ES2018, and helpers should target our minimum supported emit target.